Makefile

```diff
@@ -1,28 +1,113 @@
 PYTHON=`which python`
+PYTHON2=`which python2`
+PYTHON3=`which python3`
+PY2DSC=`which py2dsc`
+
+topdir := $(realpath $(dir $(lastword $(MAKEFILE_LIST))))
+topbuilddir := $(realpath .)
+
 DESTDIR=/
-PROJECT=python-nodefacade
+PROJECT=$(shell python $(topdir)/setup.py --name)
+VERSION=$(shell python $(topdir)/setup.py --version)
+MODNAME=$(PROJECT)
+
+# The rules for names and versions in python, rpm, and deb are different
+# and not entirely compatible. As such py2dsc will automatically convert
+# your package name into a suitable deb name and version number, and this 
+# code replicates that.
+DEBNAME=$(shell echo $(MODNAME) | tr '[:upper:]_' '[:lower:]-')
+DEBVERSION=$(shell echo $(VERSION) | sed 's/\.dev/~dev/')
+
+DEBIANDIR=$(topbuilddir)/deb_dist/$(DEBNAME)-$(DEBVERSION)/debian
+DEBIANOVERRIDES=$(patsubst $(topdir)/debian/%,$(DEBIANDIR)/%,$(wildcard $(topdir)/debian/*))
+
+RPM_PARAMS?=
+RPM_PREFIX?=$(topdir)/build/rpm
+
+RPMDIRS=BUILD BUILDROOT RPMS SOURCES SPECS SRPMS
+RPMBUILDDIRS=$(patsubst %, $(RPM_PREFIX)/%, $(RPMDIRS))
 
 all:
-	@echo "make source - Create source package"
+	@echo "$(PROJECT)-$(VERSION)"
+	@echo "make source  - Create source package"
 	@echo "make install - Install on local system (only during development)"
-	@echo "make deb - Generate a deb package - for local testing"
-	@echo "make clean - Get rid of scratch and byte files"
-	@echo "make test - Run any unit tests"
+	@echo "make clean   - Get rid of scratch and byte files"
+	@echo "make test    - Test using tox and nose2"
+	@echo "make deb     - Create deb package"
+	@echo "make dsc     - Create debian source package"
+	@echo "make rpm     - Create rpm package"
+	@echo "make wheel   - Create whl package"
+	@echo "make egg     - Create egg package"
+	@echo "make rpm_dirs - Create directories for rpm building"
 
-source:
-	$(PYTHON) setup.py sdist $(COMPILE)
+$(topbuilddir)/dist:
+	mkdir -p $@
 
-install:
-	$(PYTHON) setup.py install --root $(DESTDIR) $(COMPILE)
+source: $(topbuilddir)/dist
+	$(PYTHON) $(topdir)/setup.py sdist $(COMPILE) --dist-dir=$(topbuilddir)/dist
+
+$(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz: source
 
-deb:
-	debuild -uc -us
+install:
+	$(PYTHON) $(topdir)/setup.py install --root $(DESTDIR) $(COMPILE)
 
 clean:
-	$(PYTHON) setup.py clean
-	dh_clean
-	rm -rf build/ MANIFEST
-	find . -name '*.pyc' -delete
+	$(PYTHON) $(topdir)/setup.py clean || true
+	rm -rf $(topbuilddir)/.tox
+	rm -rf $(topbuilddir)/build/ MANIFEST
+	rm -rf $(topbuilddir)/dist
+	rm -rf $(topbuilddir)/deb_dist
+	rm -rf $(topbuilddir)/*.egg-info
+	find $(topdir) -name '*.pyc' -delete
+	find $(topdir) -name '*.py,cover' -delete
 
 test:
-	$(PYTHON) -m unittest discover -s ./tests
+	tox
+
+deb_dist: $(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz
+	$(PY2DSC) --with-python2=true --with-python3=true $(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz
```

Comment:
Does the Python 3 package actually work? See also #9 where I tried to get it working, but it's blocked on a bunch of other stuff in nmoscommon working first, and python3-pybonjour to work and python3-avahi to actually exist. Might be worth removing from here and the wheel build if it doesn't work...